### PR TITLE
recipe-persistence MySQL database implementation

### PR DESCRIPTION
**Pull request by mpalii**
An implementation of a recipe-persistence layer by MySQL database.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -14,12 +14,23 @@ subprojects {
         mavenCentral()
     }
 
+    configurations.all {
```
Comment (iherasymenko) [inline build.gradle]: `enforcedPlatform(...)` serves the same purpose.

File: settings.gradle [hunk 2]
```diff
@@ -5,4 +5,6 @@ include ':recipe-services'
 include ':recipe-persistence-in-memory'
 include ':recipe-web-ui'
 include ':recipe-web-api'
-include ':recipe-aggregator'
\ No newline at end of file
+include ':recipe-aggregator'
+include 'recipe-persistence-db-mysql'
```
Comment (iherasymenko) [inline settings.gradle]: Please follow the established conventions and do not forget about ordering. I.e. inclusion syntax and ordering.

Comment (mpalii) [reply settings.gradle]: Sorry, it means that I have to declare a 'recipe-aggregator' module (so-called "glue") at the end of the settings.gradle file?
And about synthax: you mean that I forgot a colon (':')?

Comment (iherasymenko) [reply settings.gradle]: Correct.

File: recipe-persistence-db-mysql/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
+module recipe.persistence.db_mysql {
+    requires recipe.domain;
+    requires recipe.persistence;
+    requires spring.context;
+    requires spring.core;
+    requires spring.beans;
+    requires spring.jdbc;
+    requires java.sql;
+    requires java.naming;
```
[the remark this answers is not in the document]
Comment (mpalii) [reply recipe-persistence-db-mysql/src/main/java/module-info.java]: DataSource class requires 'java.sql' module.
About 'javax.naming'.
MysqlDataSource class implements Referenceable interface from 'javax.naming' module, that is why I have to add this module to my build. I think this is the only feature of MySql.

[the remark this answers is not in the document]
Comment (iherasymenko) [reply recipe-persistence-db-mysql/src/main/java/module-info.java]: Ah, I see. We should be good then.

**Pull request closed without merging**